generator/sequence_transform_data.py

```diff
@@ -299,29 +302,84 @@ def traverse_trienode(trinode: Dict[str, Any]):
     return outputs
 
 
+###############################################################################
+def generate_matches(pattern) -> list[tuple[str, str]]:
+    square_brackets_group = re.findall(r"\[(\w+)]", pattern)
+    if square_brackets_group:
+        match = square_brackets_group[0]
+
+        return generate_matches(pattern.replace(
+            f"[{match}]",
+            f"({'|'.join(match)})"
+        ))
+
+    patterns = []
+    groups = re.findall(r"\((?:\w\|?)+\)\??", pattern)
+
+    if not groups:
+        return [("", pattern)]
+
+    match = groups[0]
+    elements = re.findall("\w+", match)
+
+    patterns.extend([
+        (element, pattern.replace(match, element))
+        for element in elements
+    ])
+
+    if "?" in match:
+        patterns.append(("", pattern.replace(match, "")))
+
+    return patterns
+
+
+###############################################################################
+def parse_tokens(tokens: List[str], parse_regex: bool) -> Iterator[Tuple[int, str, str]]:
+    full_sequence, transform = tokens
+
+    if parse_regex:
+        for token, sequence in generate_matches(full_sequence):
+            yield sequence, transform.replace(r"\1", token)
+    else:
+        yield full_sequence, transform
+
+
 ###############################################################################
 def parse_file_lines(
     file_name: str, separator: str, comment: str
 ) -> Iterator[Tuple[int, str, str]]:
     """Parses lines read from `file_name` into context-correction pairs."""
-    line_number = 0
-
     with open(file_name, 'rt', encoding="utf-8") as file:
-        for line in file:
-            line_number += 1
-            line = line.strip()
+        file_content = file.read()
 
-            if line and line.find(comment) != 0:
-                # Parse syntax "sequence -> transformation".
-                # Using strip to ignore indenting.
-                tokens = [token.strip() for token in line.split(separator, 1)]
+    lines = file_content.split("\n")
 
-                if len(tokens) != 2 or not tokens[0]:
-                    raise SystemExit(
-                        f'{err(line_number)}: Invalid syntax: "{red(line)}"'
-                    )
+    regex_start = f"{COMMENT_STR}REGEX_START"
+    regex_end = f"{COMMENT_STR}REGEX_END"
+    in_regex_zone = False
+
+    for line_number, line in enumerate(lines, 1):
+        line = line.strip()
+        if not line: continue
+
+        if line.find(regex_start) == 0:
+            in_regex_zone = True
+            continue
+        if line.find(regex_end) == 0:
+            in_regex_zone = False
+            continue
```

Comment:
I tweaked the algorithm a bit to allow regex sections to be parsed in a single pass of the file lines. A bit of a performance boost and a bit easier to follow, I think. If there is some issue I'm unaware of, we can revert